### DISCUSSION TITLE
Interval refresh sensors

### DISCUSSION THREAD

**Discussion by poudenes** [humbertogontijo/homeassistant-roborock#447] [General]
Hi All,

Is there a way to make the interval shorter for some sensors:

sensor.roborock_s8_current_room

When the vacuum is leaving a room to a other it takes some 20-30 seconds before you can see it in the sensor.
Would be great if this can be 5-10 seconds.. Or less (realtime) same as vacuum.roborock_s8. Here you can see directly when it goes cleaning, return to dock and docked.

**Comment by humbertogontijo** [comment]
30 seconds is the scan interval for entities in this integration
The status updates right away because a command was sent or roborock notified a status change. Which is not the case for room changes

**Reply by poudenes** [reply]
Can this changed? I created motion sensors that's include the room, motion sensor on sealing and status roborock. 
When roborock is in living room and motion is detected then show this new sensor not in motion. Off roborock is in hallways and there is motion in living room then the sensor shows motion.

But with a 30seconds delay it takes some time to update the current room sensor and then motion is already there before the sensor conditions are ok.

But with 30seconds this can be ok if we can't change the interval. A thief will walk around the house and then 1 or more sensors are firing.

**Comment by arne182** [comment]
What is the refresh interval for the dock errors? It looks like it only shows errors after that robot docks but in the app you see it when you change the water that the dock errors are reported directly to the app when the water tank is removed
